File: BubbleSort/bubblesort.py
def bubble_sort1(A):
    for i in range (0, len(A) - 1):
        for j in range (0, len(A) - i - 1):
            if A[j] > A[j+1]:
                A[j], A[j+1] = A[j+1], A[j]

def bubble_sort2(A):
    for i in range(0, len(A)-1):
        isSorted = true
        for j in range (0, len(A)-1):
            if A[j] > A[j+1]:
                A[j], A[j+1] = A[j+1], A[j]
                isSorted = false
            if isSorted:
                return

def bubble_sort3(A):
    for i in range(0, len(A)-1):
        isSorted = true
        for j in range (0, len(A)-1):
            if A[j] > A[j+1]:
                swap(A, j, j+1)
                isSorted = false
            if isSorted:
                return;

def swap(A, i, j):
    tmp = A[i]
    A[i] = A[j]
    A[j] = A[i]

File: BubbleSort/bubblesort.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
  public static void bubblesort(int[] array) {
    boolean isSorted = false;
    int lastUnsorted = array.length - 1;
    while (!isSorted) {
      isSorted = true;
      for (int i = 0; i < lastUnsorted - 1; i++){
        if(array[i] > array[i+1]) {
          swap(array, i, i+1);
          isSorted = false;
        }
      }
      lastUnsorted--;
    }
  }

  public static void swap(int[] array, int i, int j){
    int tmp = array[i];
    array[i] = array[j];
    array[j] = tmp;
  }
}
